### PR TITLE
Skip optimization if there are few docs

### PR DESCRIPTION
**Pull request by nik9000**
In #63643 we added an optimization to the `range` agg that uses
`filters` instead if we believe it'd be faster. It turns out that building
the `ScorerSupplier` for the `filters` has a fairly high overhead. So if
we think the query would match only a couple thousand documents per
filter we avoid the optimization alltogether. This keeps aggregations
that complete in a couple of milliseconds from wasting 50ms juggling
filters.

### CODE REVIEW

**Comment by nik9000** [comment]
Relates to https://github.com/elastic/night-rally/issues/343

File: server/src/internalClusterTest/java/org/elasticsearch/search/profile/aggregation/AggregationProfilerIT.java [hunk 1]
```diff
@@ -562,4 +567,59 @@ public void testNoProfile() {
         assertThat(profileResults, notNullValue());
         assertThat(profileResults.size(), equalTo(0));
     }
+
+    public void testFilterByFilter() throws InterruptedException, IOException {
```
Comment (not-napoleon) [inline server/src/internalClusterTest/java/org/elasticsearch/search/profile/aggregation/AggregationProfilerIT.java:571]: Comment on what this is intended to test would be nice.

**Review by not-napoleon — APPROVED** [review]
LGTM

**Comment by nik9000** [comment]
run elasticsearch-ci/packaging-sample-windows

**Pull request merged**